packages/world/ts/node/render-solidity/worldgen.ts

```diff
@@ -42,10 +43,14 @@ export async function worldgen(
         };
       }
     });
-    const systemInterfaceName = `I${system.basename}`;
+
+    const namespaceIsSet = system.basename.includes("__");
+    const { namespace, name } = resolvedConfig.systems[system.basename];
+
+    const systemInterfaceName = namespaceIsSet ? `${namespace}__I${name}` : `I${name}`;
```

Comment:
what about

```suggestion
    const systemInterfaceName = namespace === ROOT_NAMESPACE ? `I${name}` : `${namespace}__I${name}`;
```

Reply:
I agree, if the interface is user-facing we can change it to make things more consistent. 

It might be confusing though that the interface would now be named differently to your system. If you have `namespace: "app"`, and declare `SomeSystem.sol`, the interface would be `app_ISomeSystem.sol`

Reply:
By this logic, why don't we change the default behaviour of https://github.com/latticexyz/mud/pull/2880 to always use `namespaceDirectories`?

Reply:
> It might be confusing though that the interface would now be named differently to your system. If you have `namespace: "app"`, and declare `SomeSystem.sol`, the interface would be `app_ISomeSystem.sol`

to clarify, it would be `codegen/world/app__ISomeSystem.sol` (note that we're in a world codegen directory), which I think makes more sense than `codegen/world/ISomeSystem.sol` for a system defined in `app` namespace

Reply:
> By this logic, why don't we change the default behaviour of #2880 to always use `namespaceDirectories`?

because these ones _are_ imported/used in user land and generating them depends on the config alone

I don't think the generated world interfaces were ever meant to be imported in Solidity, they just separate/build up the `IWorld` interface to have an ABI that we can build and import into TS

Reply:
Thought about this more and totally agree. We don't expect users to import the interface files so changing what they are called shouldn't break any existing code. Will change the behaviour!!